flink-staging/flink-streaming/flink-streaming-core/src/main/java/org/apache/flink/streaming/api/operators/OneInputStreamOperator.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.streaming.api.operators;
+
+/**
+ * Interface for stream operators with one input. Use
+ * {@link org.apache.flink.streaming.api.operators.AbstractStreamOperator} as a base class if
+ * you want to implement a custom operator.
+ * 
+ * @param <IN> The input type of the operator
+ * @param <OUT> The output type of the operator
+ */
+public interface OneInputStreamOperator<IN, OUT> extends StreamOperator<OUT> {
+	public void receiveElement(IN element) throws Exception;
```

Comment:
`SingleInputStreamOperator` is at least in line with `SingleOutputStreamOperator`, which is fortunately a completely different thing - but the name does not sound alien.

These names are already long enough, Co is a nice and short version for TwoStream and we are already used to that.